beacon-chain/p2p/discovery.go

```diff
@@ -36,7 +39,7 @@ type Listener interface {
 // to be dynamically discoverable by others given our tracked committee ids.
 func (s *Service) RefreshENR() {
 	// return early if discv5 isnt running
-	if s.dv5Listener == nil {
+	if s.dv5Listener == nil || !s.isInitialized() {
```

Comment:
This sort of pattern is dangerous throughout the codebase because if a new person is working on something in this package, they need to remember these nil checks. Instead, maybe we could encapsulate this somewhere to prevent problems in the future. Not a big deal for this PR, but a pattern we should try to avoid moving forward. Same with the == nil || obj.IsNil() checks elsewhere

Reply:
Its a bit complicated as we have a mode which allows a node to run without any discovery. Which is why the checks are applied here as if not the method will panic. Also `s.isInitialized()` already encapsulates the state initialization into it. All ears if you have ideas on how else you want to handle this.

Reply:
I noticed there is a lot of code that interacts with the dv5listener and performs nil checks, appends, etc. Something we could do is encapsulate it and interact with it via an abstraction. That way it will make it difficult for future callers to mess things up